lib/collector/api.js

```diff
@@ -68,21 +68,28 @@ function CollectorAPI(agent) {
   /* RemoteMethods can be reused and have little per-object state, so why not
    * save some GC time?
    */
-  this._methods = {
-    preconnect: new RemoteMethod('preconnect', agent.config, initialEndpoint),
-    connect: new RemoteMethod('connect', agent.config, initialEndpoint),
-    settings: new RemoteMethod('agent_settings', agent.config, initialEndpoint),
-    errors: new RemoteMethod('error_data', agent.config, initialEndpoint),
-    metrics: new RemoteMethod('metric_data', agent.config, initialEndpoint),
-    traces: new RemoteMethod('transaction_sample_data', agent.config, initialEndpoint),
-    shutdown: new RemoteMethod('shutdown', agent.config, initialEndpoint),
-    events: new RemoteMethod('analytic_event_data', agent.config, initialEndpoint),
-    customEvents: new RemoteMethod('custom_event_data', agent.config, initialEndpoint),
-    queryData: new RemoteMethod('sql_trace_data', agent.config, initialEndpoint),
-    errorEvents: new RemoteMethod('error_event_data', agent.config, initialEndpoint),
-    spanEvents: new RemoteMethod('span_event_data', agent.config, initialEndpoint),
-    logEvents: new RemoteMethod('log_event_data', agent.config, initialEndpoint)
-  }
+  this._methods = [
+    'preconnect',
+    'connect',
+    'agent_settings',
+    'error_data',
+    'metric_data',
+    'transaction_sample_data',
+    'shutdown',
+    'analytic_event_data',
+    'custom_event_data',
+    'sql_trace_data',
+    'error_event_data',
+    'span_event_data',
+    'log_event_data'
+  ]
+    .map((methodName) => {
+      return new RemoteMethod(methodName, agent.config, initialEndpoint, agent.metrics)
+    })
+    .reduce((all, method) => {
+      all[method.name] = method
+      return all
+    }, {})
```

Comment:
I find the cognitive load of parsing the `.map(...).reduce(...)` to be slightly higher than the original copy-pastey code, but not significantly so to complain any louder than I already am.

Reply:
I just can't stand that kind of copypasta.

Reply:
I've modified the way this is done to use a for-loop instead, which should be more readable than my sicko map-reduce shenanigans.

Reply:
It's not that important to me, so sorry to ask if you changed your mind and also sorry if I'm misunderstanding, but I am not seeing the change, did you push it or am I looking at the wrong version of the code?

Reply:
I think you're looking at the wrong version of the code. Have you refreshed the page?

Reply:
I swear I reloaded the page, so anyway, yeah, it looks good now.

Reply:
Sometimes that's just how browsers are. Sorry for the trouble!